aws/macie_test.go

```diff
@@ -0,0 +1,55 @@
+package aws
+
+import (
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/macie2"
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/util"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestListMacieAccounts(t *testing.T) {
+	// Currently we hardcode to region us-east-1, because this is where our "standing" test invite exists
+	region := "us-east-1"
+	session, err := session.NewSession(&aws.Config{Region: aws.String(region)})
+	require.NoError(t, err)
+
+	accountId, err := util.GetCurrentAccountId(session)
+	require.NoError(t, err)
+
+	acceptTestInvite(t, session)
+	// Clean up after test by deleting the macie account association
+	defer nukeAllMacieMemberAccounts(session, []string{accountId})
+
+	retrievedAccountIds, lookupErr := getAllMacieMemberAccounts(session, time.Now(), config.Config{})
+	require.NoError(t, lookupErr)
+
+	assert.Contains(t, retrievedAccountIds, accountId)
+}
+
+// Macie is not very conducive to programmatic testing. In order to make this test work, we  maintain a standing invite
+// from our phxdevops test account to our nuclear-wasteland account. We can continuously "nuke" our membership because
+// Macie supports a member account *that was invited* to remove its own association at any time. Meanwhile, disassociating
+// in this manner does not destroy or invalidate the original invitation, which allows us to to continually re-accept it
+// from our nuclear-wasteland account (where cloud-nuke tests are run), just so that we can nuke it again
+//
+// Macie is also regional, so for the purposes of cost-savings and lower admin overhead, we're initially only testing this
+// in the one hardcoded region
```

Comment:
Does this have risk of breaking our macie test in the other repos? E.g., in CIS service catalog: https://github.com/gruntwork-io/terraform-aws-cis-service-catalog/blob/master/test/security/macie_test.go

Reply:
As far as I can tell, that test is using a very similar pattern: 
- The Sandbox account is the administrator account, which invites phxdevops
- Phxdevops accepts the invite, to become a member account of Sandbox 

However, Macie is regional, and the CIS service catalog test is using regions in the EU, whereas this cloud-nuke Macie test is currently hardcoded to us-east-1. 

I think that will prevent conflict - so long as neither test has its testing regions expanded.

Reply:
Added more context in 7438eea